### PR TITLE
Refactor: move some modules to main crate

### PR DESCRIPTION
**Pull request by ekexium**
This PR is based on #192 which should be merged first.

Changes are described by commit messages.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -92,15 +96,16 @@ mod proptests;
 #[macro_use]
 extern crate log;
 
+#[doc(inline)]
+pub use crate::kv::{codec, BoundRange, Key, KvPair, ToOwnedRange, Value};
```
Comment (nrc) [inline src/lib.rs:100]: Does codec need to be public?

File: src/store.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        Ok(self
+            .client
+            .dispatch(request)
+            .await?
+            .downcast()
+            .expect("Downcast failed"))
+    }
+}
+
+pub trait KvConnectStore: KvConnect {
```
Comment (nrc) [inline src/store.rs:23]: There should be an impl of this trait for `TikvConnect`

**Review by nrc — COMMENTED** [review]
one question and one missing impl, otherwise lgtm

File: src/lib.rs (hunk 1 above)
Comment (ekexium) [reply src/lib.rs:100]: For users using only this library, no. Is it possible that someone wants to manually encode keys to directly interact with PD?

Comment (nrc) [reply src/lib.rs:100]: I think that might be possible, but then it should be exported by the PD crate and not the main crate.

Comment (ekexium) [reply src/lib.rs]: Makes sense.

**Pull request merged**